crates/forge_services/src/tool_service.rs

```diff
@@ -14,32 +15,41 @@ use crate::Infrastructure;
 const TOOL_CALL_TIMEOUT: Duration = Duration::from_secs(300);
 
 #[derive(Clone)]
-pub struct ForgeToolService {
+pub struct ForgeToolService<M: ToolService + ?Sized = dyn ToolService> {
     tools: Arc<HashMap<ToolName, Tool>>,
```

Comment:
tools should represent a unified Hashmap of all the tools - The tool definition should change based on if its a MCP or a Local implementation.